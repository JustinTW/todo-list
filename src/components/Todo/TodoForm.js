import React from 'react';
import PropTypes from 'prop-types';

class TodoForm extends React.Component {
  static propTypes = {
    onSummarySubmit: PropTypes.func.isRequired,
  };

  constructor() {
    super();
    this.summaryRef = React.createRef();
    this.startRef = React.createRef();
    this.endRef = React.createRef();
    this.descriptionRef = React.createRef();
  }

  componentDidMount() {
    const start = this.startRef.current.value.trim();
    const end = this.endRef.current.value.trim();

    if (!start) {
      this.startRef.current.value = '2018-09-05T16:00:00+08:00';
    }
    if (!end) {
      this.endRef.current.value = '2018-09-05T16:30:00+08:00';
    }
  }

  doSubmit = e => {
    e.preventDefault();
    const summary = this.summaryRef.current.value.trim();
    const start = this.startRef.current.value.trim();
    const end = this.endRef.current.value.trim();
    const description = this.descriptionRef.current.value.trim();
    if (!summary || !start || !end) {
      return;
    }
    this.props.onSummarySubmit(summary, start, end, description);
    this.summaryRef.current.value = '';
    this.descriptionRef.current.value = '';
  };

  render() {
    return (
      <div className="commentForm vert-offset-top-2">
        <div className="clearfix">
          <form onSubmit={this.doSubmit}>
            <div className="form-row">
              <div className="form-group col-md-4">
                {/* eslint-disable-next-line jsx-a11y/label-has-for */}
                <label htmlFor="summary">
                  Event <span className="text-danger font-weight-bold">*</span>
                </label>
                <input
                  type="text"
                  id="summary"
                  ref={this.summaryRef}
                  className="form-control"
                  placeholder="What do you need to do?"
                  maxLength="90"
                  autoComplete="off"
                  required
                />
              </div>
              <div className="form-group col-md-3">
                {/* eslint-disable-next-line jsx-a11y/label-has-for */}
                <label htmlFor="start">
                  Start <span className="text-danger font-weight-bold">*</span>
                </label>
                <input
                  type="text"
                  id="start"
                  ref={this.startRef}
                  className="form-control"
                  placeholder="start"
                  autoComplete="off"
                  required
                />
              </div>
              <div className="form-group col-md-3">
                {/* eslint-disable-next-line jsx-a11y/label-has-for */}
                <label htmlFor="end">
                  End <span className="text-danger font-weight-bold">*</span>
                </label>
                <input
                  type="text"
                  id="end"
                  ref={this.endRef}
                  className="form-control"
                  placeholder="end"
                  autoComplete="off"
                  required
                />
              </div>
              <div className="form-group col-md-2">
                {/* eslint-disable-next-line jsx-a11y/label-has-for */}
                <label htmlFor="">Operation</label>
                <input
                  type="submit"
                  value="Add Event"
                  className="btn btn-primary float-right"
                />
              </div>
            </div>
            <div className="form-row">
              <div className="form-group col-md-10">
                {/* eslint-disable-next-line jsx-a11y/label-has-for */}
                <label htmlFor="description">Description</label>
                <input
                  type="text"
                  id="description"
                  ref={this.descriptionRef}
                  className="form-control"
                  placeholder="What is this event detail?"
                  maxLength="180"
                  autoComplete="off"
                />
              </div>
            </div>
          </form>
        </div>
        <hr />
      </div>
    );
  }
}

export default TodoForm;
