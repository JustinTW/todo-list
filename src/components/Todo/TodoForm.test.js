/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
// import shallow from 'enzyme';
import TodoForm from './TodoForm';

function createNodeMock(element) {
  if (element.type === 'input') {
    return { value: 'mock input value' };
  }
  return null;
}

describe('TodoForm', () => {
  test('renders children correctly', () => {
    const options = { createNodeMock };
    const wrapper = renderer
      .create(<TodoForm onSummarySubmit={() => {}} />, options)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
