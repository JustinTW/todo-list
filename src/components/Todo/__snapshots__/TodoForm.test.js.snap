// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm renders children correctly 1`] = `
<div
  className="commentForm vert-offset-top-2"
>
  <div
    className="clearfix"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-4"
        >
          <label
            htmlFor="summary"
          >
            Event 
            <span
              className="text-danger font-weight-bold"
            >
              *
            </span>
          </label>
          <input
            autoComplete="off"
            className="form-control"
            id="summary"
            maxLength="90"
            placeholder="What do you need to do?"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="start"
          >
            Start 
            <span
              className="text-danger font-weight-bold"
            >
              *
            </span>
          </label>
          <input
            autoComplete="off"
            className="form-control"
            id="start"
            placeholder="start"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="end"
          >
            End 
            <span
              className="text-danger font-weight-bold"
            >
              *
            </span>
          </label>
          <input
            autoComplete="off"
            className="form-control"
            id="end"
            placeholder="end"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-2"
        >
          <label
            htmlFor=""
          >
            Operation
          </label>
          <input
            className="btn btn-primary float-right"
            type="submit"
            value="Add Event"
          />
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-10"
        >
          <label
            htmlFor="description"
          >
            Description
          </label>
          <input
            autoComplete="off"
            className="form-control"
            id="description"
            maxLength="180"
            placeholder="What is this event detail?"
            type="text"
          />
        </div>
      </div>
    </form>
  </div>
  <hr />
</div>
`;
